src/privatesend/privatesend-client.cpp

```diff
@@ -631,6 +631,30 @@ bool CPrivateSendClientSession::SignFinalTransaction(const CTransaction& finalTr
     std::vector<CTxIn> sigs;
 
     for (const auto& entry : vecEntries) {
+        // Check that the final transaction has all our outputs
+        int nOutputsCountFinal = 0;
+        int nOuputsCountTarget = entry.vecTxOut.size();
+
+        for (const auto& txout : entry.vecTxOut) {
+            for (const auto& txoutFinal : finalMutableTransaction.vout) {
+                if (txoutFinal == txout) {
+                    nOutputsCountFinal++;
+                    break;
+                }
+            }
+        }
+
+        if (nOutputsCountFinal < nOuputsCountTarget) {
```

Comment:
Can we be 100% sure that the `nValue1 != nValue2` isn't needed here?

Reply:
Yes, because we already checked denoms above. But in any case, you can ignore this part here because it is replaced with a better version (where we actually look for the exact output match) in the following commit.